go/extractor/autobuilder/build-environment.go

```diff
@@ -82,57 +80,57 @@ func getVersionWhenGoModVersionNotFound(v versionInfo) (msg, version string) {
 
 // Assuming `v.goModVersion` is above the supported range, emit a diagnostic and return the
 // version to install, or the empty string if we should not attempt to install a version of Go.
-func getVersionWhenGoModVersionTooHigh(v versionInfo) (msg, version string) {
-	if !v.goEnvVersionFound {
+func getVersionWhenGoModVersionTooHigh(v versionInfo) (msg string, version util.SemVer) {
+	if v.goEnvVersion == nil {
 		// The version in the `go.mod` file is above the supported range. There is no Go version
 		// installed. We install the maximum supported version as a best effort.
-		msg = "The version of Go found in the `go.mod` file (" + v.goModVersion +
-			") is above the supported range (" + minGoVersion + "-" + maxGoVersion +
+		msg = "The version of Go found in the `go.mod` file (" + v.goModVersion.String() +
```

Comment:
(musing) It's a shame that we have to have all these `.String()`s everywhere. If we were printing directly we could use `%s` and the `String()` method would automatically be called on the argument (I believe). But that would involve major refactoring.

Reply:
Yeah, I agree. While I was doing this I thought about replacing all of this with format strings. However, that seemed a bit unrelated to the work in this PR and I didn't want to make the diff even worse than it already is. We could look at doing that separately, though.